tools/joffreBulkBooking.js

```diff
@@ -0,0 +1,139 @@
+const AWS = require('aws-sdk');
+const readline = require('readline');
+
+const TABLE_NAME = process.env.TABLE_NAME || 'parksreso';
+
+const options = {
+  region: 'ca-central-1',
+  // endpoint: 'http://localhost:8000'
+};
+
+// todo: need to confirm correct values for parkName, facilityName, email, firstName, lastName
+const parkName = 'Joffre Lakes Provincial Park';
+const facilityName = 'Day-Use Trails';
+const email = 'parkinfo@gov.bc.ca';
+const firstName = 'Bulk';
+const lastName = 'Pre-booking';
+const facilityType = 'Trail';
+const type = 'DAY';
+const passStatus = 'reserved';
+
+let passDate, shortPassDate, numberOfGuests, registrationNumber;
+
+const dynamodb = new AWS.DynamoDB(options);
+
+const rl = readline.createInterface({
+  input: process.stdin,
+  output: process.stdout
+});
+
+console.log('\nJOFFRE LAKES BULK BOOKING TOOL\n');
+console.log('Warning: This tool writes directly to the AWS database with very minimal input validation.');
+console.log('Please ensure your inputs are valid and read JOFFRE-README.md before using this tool.');
+console.log('Press CTRL-C to exit.\n');
+console.log(`Park name: ${parkName}`);
+console.log(`Facility name: ${facilityName}`);
+console.log(`Pass type: ${type}`);
+
+rl.question('Booking date (YYYY-MM-DD): ', async function (dateInput) {
+  shortPassDate = dateInput;
+
+  // note: the date is being set to 12 noon UTC time to match the Angular FE app.
+  // Any updates to how dates are recorded in the Angular app should also be made here!
+  passDate = new Date(Date.parse(shortPassDate) + 12 * 60 * 60 * 1000);
+
+  // make the registration number non-random so the sk on the pass is repeatable
+  // and we can overwrite records in the db with a put
+  registrationNumber = '00' + shortPassDate.replace(/-/g, '');
+
+  rl.question('Number of guests?: ', async function (guestsInput) {
+    numberOfGuests = guestsInput;
```

Comment:
We should be checking the current & allowed capacities of the facility and preventing the user from overbooking.

Reply:
This shouldn't really be an issue since we don't allow the script to be run once the booking windows has opened.  
e.g. Joffre has 1053 passes per day, and the script will only be used to book 54 passes.  If they wanted to book 1100 passes and sell out the whole day, then there is a probably a reason for it.